### PR TITLE
arch: arm: aarch32: add Kconfig for arm cortex-m that implements a cache

### PR DESCRIPTION
**Pull request by XenuIsWatching**
The Cache is an optional configuration of both the ARM Cortex-M7 and
Cortex-M55. Previously, it was just checking that it was just an M7
rather than knowing that the CPU actually was built with the cache.

Signed-off-by: Ryan McClelland <ryanmcclelland@fb.com>

### CODE REVIEW

**Review by MaureenHelm — DISMISSED** [review]

**Review by microbuilder — DISMISSED** [review]

File: arch/arm/core/aarch32/cortex_m/scb.c [hunk 1]
```diff
@@ -108,7 +108,7 @@ void z_arm_init_arch_hw_at_boot(void)
 		NVIC->ICPR[i] = 0xFFFFFFFF;
 	}
 
-#if defined(CONFIG_CPU_CORTEX_M7)
+#if defined(CPU_CORTEX_M_HAS_CACHE)
```
Comment (microbuilder) [inline arch/arm/core/aarch32/cortex_m/scb.c:111]:
```suggestion
#if defined(CONFIG_CPU_CORTEX_M_HAS_CACHE)
```

File: arch/arm/core/aarch32/cortex_m/scb.c [hunk 2]
```diff
@@ -121,7 +121,7 @@ void z_arm_init_arch_hw_at_boot(void)
 	}
 	/* Reset I-Cache settings. */
 	SCB_DisableICache();
-#endif /* CONFIG_CPU_CORTEX_M7 */
+#endif /* CPU_CORTEX_M_HAS_CACHE */
```
Comment (microbuilder) [inline arch/arm/core/aarch32/cortex_m/scb.c:124]:
```suggestion
#endif /* CONFIG_CPU_CORTEX_M_HAS_CACHE */
```

**Review by microbuilder — CHANGES_REQUESTED** [review]

**Comment by XenuIsWatching** [comment]
I'm not sure what's going on with the "Cancelled check". I'll rebase and push again with no changes.

**Review by MaureenHelm — DISMISSED** [review]

**Review by microbuilder — APPROVED** [review]

**Comment by XenuIsWatching** [comment]
This is also now adding defines for DCACHE and ICACHE present for the MP3 SE300 which has a 64kB cache for each.

**Review by MaureenHelm — APPROVED** [review]

File: soc/arm/st_stm32/stm32h7/Kconfig.soc [hunk 3]
```diff
@@ -12,16 +12,19 @@ choice
...
+	select CPU_CORTEX_M_HAS_CACHE
 	select CPU_HAS_FPU_DOUBLE_PRECISION
 
 config SOC_STM32H735XX
 	bool "STM32H735XX"
 	select CPU_CORTEX_M7
+	select CPU_CORTEX_M_HAS_CACHE
 	select CPU_HAS_FPU_DOUBLE_PRECISION
 
 config SOC_STM32H743XX
```
[the remark this answers is not in the document]
Comment (erwango) [reply soc/arm/st_stm32/stm32h7/Kconfig.soc:30]: Yes, I've a fix on going: https://github.com/zephyrproject-rtos/zephyr/pull/45109

[the remark this answers is not in the document]
Comment (XenuIsWatching) [reply soc/arm/st_stm32/stm32h7/Kconfig.soc:30]: opps... :(
I dunno how I missed that.

I did a quick search (to triple check if I missed anything else) and I found another line in the mcuboot that needs to change over to use the new KConfig

**Pull request merged**